pkg/cmd/cli/cmd/rollback.go

```diff
@@ -0,0 +1,97 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	kubecmd "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd"
+
+	latestapi "github.com/openshift/origin/pkg/api/latest"
+	client "github.com/openshift/origin/pkg/cmd/client"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+const rollbackLongDesc = `
+Rolls back a deploymentConfig to match a previous deployment.
+
+By default, only the pod template of the previous deployment is carried over for the
+rollback. Other aspects of the previous deployment, such as triggers and the previous
+strategy used must be explicitly selected for inclusion during the rollback.
+
+When the 'output' option is specified, the JSON of the deploymentConfig for the rollback
+will be printed, but no rollback will actually be performed. This is useful for testing,
+dry runs, and chaining with other commands.
+
+Examples:
+  $ %[1]s %[2]s deployment-2
+  ...
+
+  $ %[1]s %[2]s deployment-3 --include-strategy=true
+  ...
+
+  $ %[1]s %[2]s deployment-1 --output=json
+  ...
+
+  $ %[1]s %[2]s deployment-1 --output=json | %[1]s update deploymentConfigs deployment -f -
+`
+
+func NewCmdRollback(parentName string, name string, f *Factory, out io.Writer) *cobra.Command {
+	rollback := &deployapi.DeploymentConfigRollback{
+		Spec: deployapi.DeploymentConfigRollbackSpec{
+			IncludeTemplate: true,
+		},
+	}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s <from-deployment>", name),
+		Short: "Rolls back a deploymentConfig to match a previous deployment.",
+		Long:  fmt.Sprintf(rollbackLongDesc, parentName, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 || len(args[0]) == 0 {
+				usageError(cmd, "A deployment name is required.")
+			}
+
+			rollback.Spec.From.Name = args[0]
+
+			output := kubecmd.GetFlagString(cmd, "output")
+
+			osClient, _, err := f.Clients(cmd)
+			checkErr(err)
+
+			namespace := getOriginNamespace(cmd)
+
+			newConfig, err := osClient.DeploymentConfigs(namespace).Rollback(rollback)
+			checkErr(err)
+
+			// If an output format is specified, display the rollback config JSON and exit
+			// WITHOUT performing a rollback.
+			if len(output) > 0 {
+				switch output {
```

Comment:
You really should be supporting the full output options here, including Describe output.